### PR TITLE
Zero initialization as beta=0 during fusion

### PR DESCRIPTION
**Pull request by KavithaTipturMadhu**
https://github.com/plaidml/tpp-mlir/pull/897#discussion_r1512447709

### CODE REVIEW

File: lib/TPP/Transforms/LinalgConvertCompareSelectToMaximumfPass.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                                PatternRewriter &rewriter) const override {
+
+    if (op.getBody()->getOperations().size() != 3 ||
+        !(dyn_cast<arith::CmpFOp>(&op.getBody()->getOperations().front()) &&
+          (dyn_cast<arith::SelectOp>(
+              std::next(op.getBody()->getOperations().begin(), 1))))) {
+      return failure();
+    }
+
+    auto cmpf = dyn_cast<arith::CmpFOp>(&op.getBody()->getOperations().front());
```
Comment (adam-smnk) [inline lib/TPP/Transforms/LinalgConvertCompareSelectToMaximumfPass.cpp:44]: You should also check that the predicate is correct and add some negative test cases for that

**Review by adam-smnk — COMMENTED** [review]
nit: the PR is very much "FC" ;)

**Comment by adam-smnk** [comment]
Any impact on benchmarks?

File: lib/TPP/DefaultTppPasses.cpp [hunk 2]
```diff
@@ -217,7 +217,10 @@ struct TppMapping : public tpp::impl::TppMappingBase<TppMapping>,
     pm.addPass(createConstantFoldPack());
     pm.addPass(createSimplifyAndCanonicalizePack());
 
+    pm.addNestedPass<func::FuncOp>(createLinalgGeneralizeNamedOpsPass());
```
[the remark this answers is not in the document]
Comment (KavithaTipturMadhu) [reply lib/TPP/DefaultTppPasses.cpp]: sorry this patch was rebased on top of my other branch which wasn't necessary. I have rebased on top of main now, should have just the necessary changes for review.

File: benchmarks/mlir/pytorch/torch-dynamo-mlp-bf16-3x1024.mlir
[the remark this answers is not in the document]
Comment (KavithaTipturMadhu) [reply benchmarks/mlir/pytorch/torch-dynamo-mlp-bf16-3x1024.mlir:1]: spurious changes, removed.

File: lib/TPP/Transforms/LinalgConvertCompareSelectToMaximumfPass.cpp (hunk 1 above)
Comment (KavithaTipturMadhu) [reply lib/TPP/Transforms/LinalgConvertCompareSelectToMaximumfPass.cpp]: spurious changes, removed.

**Comment by KavithaTipturMadhu** [comment]
> Any impact on benchmarks?

Will have impact on benchmarks and tests, will add some unit tests at least.

File: lib/TPP/Dialect/Xsmm/XsmmUtils.cpp [hunk 3]
```diff
@@ -335,6 +340,12 @@ FailureOr<FusedMatch> getFusedBrgemmSequenceFromProducer(Operation *op) {
   //  - GEMM + BINARY + UNARY
```
Comment (adam-smnk) [inline lib/TPP/Dialect/Xsmm/XsmmUtils.cpp:340]: Some comments (like this one) are a bit outdated with this change.

**Review by rengolin — COMMENTED** [review]
This is really good, thank you @KavithaTipturMadhu, and sorry for the delay. This week is being a bit tough.

I only have two nits and it should be good to merge.

**Review by rengolin — APPROVED** [review]

**Pull request merged**